instance/models/mixins/secret_keys.py

```diff
@@ -83,6 +98,10 @@ def get_key_provider(self):
         Get a proxy we can pass to a template that'll generate secret keys
         for any attribute requested from it.
         """
+        if not self.apserver_set.count() and not self.secret_key_b64encoded:
```

Comment:
Nit (no need to change, but for future reference): better to write `if not self.appserver_set.exists()` (more efficient) or `if count() == 0` (more clear)